core/src/blocks/web_scrape.rs

```diff
@@ -0,0 +1,149 @@
+use crate::blocks::block::{parse_pair, replace_variables_in_string, Block, BlockType, Env};
+use crate::http::request::HttpRequest;
+use crate::Rule;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use pest::iterators::Pair;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Error {
+    pub error: String,
+}
+
+#[derive(Clone)]
+pub struct Web_Scrape {
+    url: String,
+    selector: String,
+}
+
+impl Web_Scrape {
+    pub fn parse(block_pair: Pair<Rule>) -> Result<Self> {
+        let mut url: Option<String> = None;
+        let mut selector: Option<String> = None;
+
+        for pair in block_pair.into_inner() {
+            match pair.as_rule() {
+                Rule::pair => {
+                    let (key, value) = parse_pair(pair)?;
+                    match key.as_str() {
+                        "url" => url = Some(value),
+                        "selector" => selector = Some(value),
+                        _ => Err(anyhow!("Unexpected `{}` in `web_scrape` block", key))?,
+                    }
+                }
+                Rule::expected => {
+                    Err(anyhow!("`expected` is not yet supported in `web_scrape` block"))?
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        if !url.is_some() {
+            Err(anyhow!("Missing required `url` in `web_scrape` block"))?;
+        }
+
+        Ok(Web_Scrape {
+            url: url.unwrap(),
+            selector: match selector {
+                Some(selector) => selector,
+                None => "body".to_string(),
+            },
+        })
+    }
+}
+
+
+#[async_trait]
+impl Block for Web_Scrape {
```

Comment:
For the record, it's called `Web_Scrape` for consistency with `BlockType::Web_Scrape`, and that uses an underscord because the API fails if it's called `BlockType::WebScrape`. The way the serialization works, the block type names have to be the same as the lowercased version of the name of the block type in the spec. There may be a way to override the serialization to make this not necessary, but I didn't know enough about the API framework to figure it out.

In any case, it's moot if we change to a one word name like `Browser`, but it's something we'll run into again with two word block types.

Reply:
And `Browser`/`browser` sounds good to me!